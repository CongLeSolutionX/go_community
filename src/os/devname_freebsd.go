// Copyright 2022 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package os

import (
	"internal/syscall/unix"
	"syscall"
	"unsafe"
)

// devname returns the name of the block or character device in /dev with a device number of dev.
// To find the right name, devname asks the kernel via the "kern.devname" sysctl.
func devname(dev uint64) (string, error) {
	mib, nerr := unix.SysctlNameToMIB([]byte("kern.devname"))
	if nerr != nil {
		return "", nerr
	}
	miblen := uintptr(len(mib))

	// get name length
	n := uintptr(0)
	_, _, err := syscall.Syscall6(syscall.SYS___SYSCTL, uintptr(unsafe.Pointer(&mib[0])), miblen, 0, uintptr(unsafe.Pointer(&n)), uintptr(unsafe.Pointer(&dev)), unsafe.Sizeof(dev))
	if err != 0 {
		return "", err
	}
	if n == 0 {
		return "", nil // shouldn't happen
	}
	// get name
	buf := make([]byte, n)
	_, _, err = syscall.Syscall6(syscall.SYS___SYSCTL, uintptr(unsafe.Pointer(&mib[0])), miblen, uintptr(unsafe.Pointer(&buf[0])), uintptr(unsafe.Pointer(&n)), uintptr(unsafe.Pointer(&dev)), unsafe.Sizeof(dev))
	if err != 0 {
		return "", err
	}
	if n == 0 {
		return "", nil // shouldn't happen
	}
	return string(buf[:n-1]), nil
}
