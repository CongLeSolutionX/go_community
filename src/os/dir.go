// Copyright 2016 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package os

// ReadDir reads the contents of the directory associated with file and
// returns a slice of up to n FileInfo values, as would be returned
// by Lstat, in directory order. Subsequent calls on the same file will yield
// further FileInfos.
//
// If n > 0, ReadDir returns at most n FileInfo structures. In this case, if
// ReadDir returns an empty slice, it will return a non-nil error
// explaining why. At the end of a directory, the error is io.EOF.
//
// If n <= 0, ReadDir returns all the FileInfo from the directory in
// a single slice. In this case, if ReadDir succeeds (reads all
// the way to the end of the directory), it returns the slice and a
// nil error. If it encounters an error before the end of the
// directory, ReadDir returns the FileInfo read until that point
// and a non-nil error.
func (f *File) ReadDir(n int) ([]FileInfo, error) {
	if f == nil {
		return nil, ErrInvalid
	}
	return f.readdir(n)
}

// Readdir is an old name for ReadDir.
func (f *File) Readdir(n int) ([]FileInfo, error) {
	return f.ReadDir(n)
}

// ReadDirNames reads the contents of the directory associated with file
// and returns a slice of up to n names of files in the directory,
// in directory order. Subsequent calls on the same file will yield
// further names.
//
// If n > 0, ReadDirNames returns at most n names. In this case, if
// ReadDirNames returns an empty slice, it will return a non-nil error
// explaining why. At the end of a directory, the error is io.EOF.
//
// If n <= 0, ReadDirNames returns all the names from the directory in
// a single slice. In this case, if ReadDirNames succeeds (reads all
// the way to the end of the directory), it returns the slice and a
// nil error. If it encounters an error before the end of the
// directory, ReadDirNames returns the names read until that point and
// a non-nil error.
func (f *File) ReadDirNames(n int) (names []string, err error) {
	if f == nil {
		return nil, ErrInvalid
	}
	return f.readdirnames(n)
}

// Readdirnames is an old name for ReadDirNames.
func (f *File) Readdirnames(n int) ([]string, error) {
	return f.ReadDirNames(n)
}
