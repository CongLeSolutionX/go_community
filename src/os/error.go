// Copyright 2009 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package os

import (
	"errors"
	"syscall"
)

// Portable analogs of some common system call errors.
var (
	ErrInvalid    = errors.New("invalid argument")
	ErrPermission = errors.New("permission denied")
	ErrExist      = errors.New("file already exists")
	ErrNotExist   = errors.New("file does not exist")
)

// PathError records an error and the operation and file path that caused it.
type PathError struct {
	Op   string
	Path string
	Err  error
}

func (e *PathError) Error() string { return e.Op + " " + e.Path + ": " + e.Err.Error() }

// SyscallError records an error from a specific system call.
type SyscallError struct {
	Syscall string
	Err     error
}

func (e *SyscallError) Error() string { return e.Syscall + ": " + e.Err.Error() }

// NewSyscallError returns, as an error, a new SyscallError
// with the given system call name and error details.
// As a convenience, if err is nil, NewSyscallError returns nil.
func NewSyscallError(syscall string, err error) error {
	if err == nil {
		return nil
	}
	return &SyscallError{syscall, err}
}

// IsExist returns a boolean indicating whether the error is known to report
// that a file or directory already exists. It is satisfied by ErrExist as
// well as some syscall errors.
func IsExist(err error) bool {
	return isExist(err)
}

// IsNotExist returns a boolean indicating whether the error is known to
// report that a file or directory does not exist. It is satisfied by
// ErrNotExist as well as some syscall errors.
func IsNotExist(err error) bool {
	return isNotExist(err)
}

// IsPermission returns a boolean indicating whether the error is known to
// report that permission is denied. It is satisfied by ErrPermission as well
// as some syscall errors.
func IsPermission(err error) bool {
	return isPermission(err)
}

// Ugly hack to avoid allocations when comparing type error (an interface)
// to type syscall.Errno (a uintptr). The allocation would come from
// runtime.convT2I, inserted by the compiler in an expr like
// "if err == syscall.EPIPE", in order to promote syscall.EPIPE to an
// interface for the comparision.
//
// A fix for issue 8618 would probably remove the need for this.
func errEq(e1 error, e2 syscall.Errno) bool {
	//	e3, ok := e1.(syscall.Errno)
	//	return ok && e2 == e3
	return e1 == e2
}
