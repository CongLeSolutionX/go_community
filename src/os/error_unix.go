// Copyright 2009 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris

package os

import "syscall"

func isExist(err error) bool {
	if err == nil {
		return false
	}
	err = convertError(err)
	return err == syscall.EEXIST || err == syscall.ENOTEMPTY || err == ErrExist
}

func isNotExist(err error) bool {
	if err == nil {
		return false
	}
	err = convertError(err)
	return err == syscall.ENOENT || err == ErrNotExist
}

func isPermission(err error) bool {
	if err == nil {
		return false
	}
	err = convertError(err)
	return err == syscall.EACCES || err == syscall.EPERM || err == ErrPermission
}
