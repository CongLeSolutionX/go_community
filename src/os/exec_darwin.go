// Copyright 2019 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package os

/*
#include <sys/proc.h>
*/
import "C"

import (
	"errors"
	"syscall"
	"unsafe"
)

type kProc C.struct_extern_proc

func findProcess(pid int) (p *Process, err error) {
	args := [4]int32{1 /* CTL_KERN */, 14 /* KERN_PROC */, 1 /* KERN_PROC_PID */, int32(pid)}
	args_len := uintptr(len(args))

	// Get the size first.
	size := uintptr(0)
	_, _, errno := syscall.Syscall6(
		syscall.SYS___SYSCTL,
		uintptr(unsafe.Pointer(&args[0])),
		args_len,
		0,
		uintptr(unsafe.Pointer(&size)),
		0,
		0,
	)
	if errno != 0 {
		return nil, errno
	}
	buf := make([]byte, size)
	_, _, errno = syscall.Syscall6(
		syscall.SYS___SYSCTL,
		uintptr(unsafe.Pointer(&args[0])),
		args_len,
		uintptr(unsafe.Pointer(&buf[0])),
		uintptr(unsafe.Pointer(&size)),
		0,
		0,
	)
	if errno != 0 {
		return nil, errno
	}
	kp := (*kProc)(unsafe.Pointer(&buf[0]))
	if kp.p_pid == 0 {
		return nil, ErrProcessNotExist
	}
	if int(kp.p_pid) != pid {
		// Impossible but should be reported nonetheless.
		return nil, errors.New("Pid mismatch")
	}
	return newProcess(pid, 0), nil
}
