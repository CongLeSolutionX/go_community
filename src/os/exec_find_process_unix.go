// +build aix dragonfly freebsd js,wasm linux netbsd openbsd solaris

// Copyright 2019 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package os

func findProcess(pid int) (p *Process, err error) {
	// NOOP for unix.
	return newProcess(pid, 0), nil
}
