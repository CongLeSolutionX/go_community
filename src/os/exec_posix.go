// Copyright 2009 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

//go:build unix || (js && wasm) || wasip1 || windows

package os

import (
	"internal/itoa"
	"internal/syscall/execenv"
	"runtime"
	"syscall"
)

// The only signal values guaranteed to be present in the os package on all
// systems are os.Interrupt (send the process an interrupt) and os.Kill (force
// the process to exit). On Windows, sending os.Interrupt to a process with
// os.Process.Signal is not implemented; it will return an error instead of
// sending a signal.
var (
	Interrupt Signal = syscall.SIGINT
	Kill      Signal = syscall.SIGKILL
)

func startProcess(name string, argv []string, attr *ProcAttr) (p *Process, err error) {
	// If there is no SysProcAttr (ie. no Chroot or changed
	// UID/GID), double-check existence of the directory we want
	// to chdir into. We can make the error clearer this way.
	if attr != nil && attr.Sys == nil && attr.Dir != "" {
		if _, err := Stat(attr.Dir); err != nil {
			pe := err.(*PathError)
			pe.Op = "chdir"
			return nil, pe
		}
	}

	sysattr := &syscall.ProcAttr{
		Dir: attr.Dir,
		Env: attr.Env,
		Sys: attr.Sys,
	}
	if sysattr.Env == nil {
		sysattr.Env, err = execenv.Default(sysattr.Sys)
		if err != nil {
			return nil, err
		}
	}
	sysattr.Files = make([]uintptr, 0, len(attr.Files))
	for _, f := range attr.Files {
		sysattr.Files = append(sysattr.Files, f.Fd())
	}

	pid, h, e := syscall.StartProcess(name, argv, sysattr)

	// Make sure we don't run the finalizers of attr.Files.
	runtime.KeepAlive(attr)

	if e != nil {
		return nil, &PathError{Op: "fork/exec", Path: name, Err: e}
	}

	return newProcess(pid, h), nil
}

func (p *Process) kill() error {
	return p.Signal(Kill)
}

// ProcessState stores information about a process, as reported by Wait.
type ProcessState struct {
	pid     int     // The process's id.
	siginfo Siginfo // System-dependent status info.
	rusage  *syscall.Rusage
}

// Pid returns the process id of the exited process.
func (p *ProcessState) Pid() int {
	return p.pid
}

func (p *ProcessState) exited() bool {
	return p.siginfo.Exited()
}

func (p *ProcessState) success() bool {
	return p.siginfo.ExitStatus() == 0
}

func (p *ProcessState) sys() any {
	return p.siginfo
}

func (p *ProcessState) sysUsage() any {
	return p.rusage
}

func (p *ProcessState) String() string {
	if p == nil {
		return "<nil>"
	}
	siginfo := p.Sys().(Siginfo)
	res := ""
	switch {
	case siginfo.Exited():
		code := siginfo.ExitStatus()
		if runtime.GOOS == "windows" && uint(code) >= 1<<16 { // windows uses large hex numbers
			res = "exit status " + uitox(uint(code))
		} else { // unix systems use small decimal integers
			res = "exit status " + itoa.Itoa(code) // unix
		}
	case siginfo.Signaled():
		res = "signal: " + siginfo.Signal().String()
	case siginfo.Stopped():
		res = "stop signal: " + siginfo.StopSignal().String()
		if siginfo.Trapped() && siginfo.TrapCause() != 0 {
			res += " (trap " + itoa.Itoa(siginfo.TrapCause()) + ")"
		}
	case siginfo.Continued():
		res = "continued"
	}
	if siginfo.CoreDump() {
		res += " (core dumped)"
	}
	return res
}

// ExitCode returns the exit code of the exited process, or -1
// if the process hasn't exited or was terminated by a signal.
func (p *ProcessState) ExitCode() int {
	// return -1 if the process hasn't started.
	if p == nil {
		return -1
	}
	return p.siginfo.ExitStatus()
}
