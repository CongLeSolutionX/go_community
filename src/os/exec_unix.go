// Copyright 2009 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

//go:build unix || (js && wasm) || wasip1

package os

import (
	"errors"
	"runtime"
	"syscall"
	"time"
)

func (p *Process) wait() (ps *ProcessState, err error) {
	if p.handle == nil {
		return nil, syscall.EINVAL
	}
	var (
		status syscall.WaitStatus
		rusage syscall.Rusage
	)
	if e := p.handle.Wait4(&status, &rusage); e != nil {
		return nil, NewSyscallError("wait", e)
	}
	ps = &ProcessState{
		pid:    p.Pid,
		status: status,
		rusage: &rusage,
	}
	return ps, nil
}

func (p *Process) signal(sig Signal) error {
	if p.Pid == -1 {
		return errors.New("os: process already released")
	}
	if p.Pid == 0 {
		return errors.New("os: process not initialized")
	}
	if p.handle != nil {
		if !p.handle.Hold() {
			return ErrProcessDone
		}
		defer p.handle.Unhold()
	}
	s, ok := sig.(syscall.Signal)
	if !ok {
		return errors.New("os: unsupported signal type")
	}
	if e := syscall.Kill(p.Pid, s); e != nil {
		if e == syscall.ESRCH {
			return ErrProcessDone
		}
		return e
	}
	return nil
}

func (p *Process) release() error {
	// NOOP for unix.
	p.Pid = -1
	p.handle = nil
	// no need for a finalizer anymore
	runtime.SetFinalizer(p, nil)
	return nil
}

func findProcess(pid int) (p *Process, err error) {
	// NOOP for unix.
	return newProcess(pid, nil), nil
}

func (p *ProcessState) userTime() time.Duration {
	return time.Duration(p.rusage.Utime.Nano()) * time.Nanosecond
}

func (p *ProcessState) systemTime() time.Duration {
	return time.Duration(p.rusage.Stime.Nano()) * time.Nanosecond
}
