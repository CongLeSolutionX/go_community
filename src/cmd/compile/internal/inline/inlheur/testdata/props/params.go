// Copyright 2023 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package params

import "os"

var G int

type T int

type I interface {
	Blarg()
}

func (r T) Blarg() {
}

func T_feeds_if_simple(x int) {
	if x < 100 {
		os.Exit(1)
	}
	println(x)
}

func T_feeds_if_pointer(xp *int) {
	if xp != nil {
		os.Exit(1)
	}
	println(xp)
}

func (r T) T_feeds_if_simple_method(x int) {
	if x < 100 {
		os.Exit(1)
	}
	if r != 99 {
		os.Exit(2)
	}
	println(x)
}

func T_feeds_if_blanks(_ string, x int, _ bool, _ bool) {
	if x < 100 {
		os.Exit(1)
	}
	println(x)
}

// simple copy here -- we get this case
func T_feeds_if_with_copy(x int) {
	xx := x
	if xx < 100 {
		os.Exit(1)
	}
	println(x)
}

// this case (copy of expression) currently not handled.
func T_feeds_if_with_copy_expr(x int) {
	xx := x < 100
	if xx {
		os.Exit(1)
	}
	println(x)
}

func T_feeds_switch(x int) {
	switch x {
	case 101:
		println(101)
	case 202:
		panic("bad")
	}
	println(x)
}

// not handled at the moment; we only look for cases where
// an "if" or "switch" can be simplified based on a single
// constant param, not a combination of constant params.
func T_feeds_if_toocomplex(x int, y int) {
	if x < y {
		panic("bad")
	}
	println(x + y)
}

func T_feeds_if_redefined(x int) {
	if x < G {
		x++
	}
	if x == 101 {
		panic("bad")
	}
}

// this currently classifies "x" as "no info", since the analysis we
// use to check for reassignments/redefinitions is not flow-sensitive,
// but we could probably catch this case with better analysis or
// high-level SSA.
func T_feeds_if_redefined2(x int) {
	if x == 101 {
		panic("bad")
	}
	if x < G {
		x++
	}
}

// Here we have one "if" that is too complex (x < y) but one that is
// simple enough. Currently we enable the heuristic for this. It's
// possible to imagine this being a bad thing if the function in
// question is sufficiently large, but if it's too large we probably
// can't inline it anyhow.
func T_feeds_multi_if(x int, y int) {
	if x < y {
		panic("bad")
	}
	if x < 10 {
		panic("whatev")
	}
	println(x + y)
}

func T_feeds_if_redefined_indirectwrite(x int) {
	ax := &x
	if G != 2 {
		*ax = G
	}
	if x == 101 {
		panic("bad")
	}
}

// we don't catch this case, "x" is marked as no info,
// since we're conservative about redefinitions.
func T_feeds_if_redefined_indirectwrite_copy(x int) {
	ax := &x
	cx := x
	if G != 2 {
		*ax = G
	}
	if cx == 101 {
		panic("bad")
	}
}

func T_feeds_if_expr1(x int) {
	if x == 101 || x == 102 || x&0xf == 0 {
		panic("bad")
	}
}

func T_feeds_if_expr2(x int) {
	if (x*x)-(x+x)%x == 101 || x&0xf == 0 {
		panic("bad")
	}
}

func T_feeds_if_expr3(x int) {
	if x-(x&0x1)^378 > (1 - G) {
		panic("bad")
	}
}

// here if "x" is a constant like 2, we could simplify the "if",
// but if we were to pass in a negative value for "x" we can't
// fold the condition due to the need to panic on negative shift.
func T_feeds_if_shift_may_panic(x int) *int {
	if 1<<x > 1024 {
		return nil
	}
	return &G
}

func T_feeds_if_maybe_divide_by_zero(x int) {
	if 99/x == 3 {
		return
	}
	println("blarg")
}

func T_feeds_indcall(x func()) {
	if G != 20 {
		x()
	}
}

func T_feeds_indcall_and_if(x func()) {
	if x != nil {
		x()
	}
}

func T_feeds_indcall_with_copy(x func()) {
	xx := x
	if G < 10 {
		G--
	}
	xx()
}

func T_feeds_interface_method_call(i I) {
	i.Blarg()
}
