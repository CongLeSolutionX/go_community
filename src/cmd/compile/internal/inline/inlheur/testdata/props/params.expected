// DO NOT EDIT (use 'go test -update-expected' instead))
T.Blarg
===
{
	"Flags": 0,
	"RecvrParamFlags": [
		0
	],
	"ReturnFlags": []
}
=-=-=-
T_feeds_if_simple
// RecvrParamFlags:
//   0: ParamFeedsIfOrSwitch
===
{
	"Flags": 0,
	"RecvrParamFlags": [
		8
	],
	"ReturnFlags": []
}
=-=-=-
T_feeds_if_pointer
// RecvrParamFlags:
//   0: ParamFeedsIfOrSwitch
===
{
	"Flags": 0,
	"RecvrParamFlags": [
		8
	],
	"ReturnFlags": []
}
=-=-=-
T.T_feeds_if_simple_method
// RecvrParamFlags:
//   0: ParamFeedsIfOrSwitch
//   1: ParamFeedsIfOrSwitch
===
{
	"Flags": 0,
	"RecvrParamFlags": [
		8,
		8
	],
	"ReturnFlags": []
}
=-=-=-
T_feeds_if_blanks
// RecvrParamFlags:
//   0: ParamFeedsIfOrSwitch
===
{
	"Flags": 0,
	"RecvrParamFlags": [
		8
	],
	"ReturnFlags": []
}
=-=-=-
T_feeds_if_with_copy
// RecvrParamFlags:
//   0: ParamFeedsIfOrSwitch
===
{
	"Flags": 0,
	"RecvrParamFlags": [
		8
	],
	"ReturnFlags": []
}
=-=-=-
T_feeds_if_with_copy_expr
===
{
	"Flags": 0,
	"RecvrParamFlags": [
		0
	],
	"ReturnFlags": []
}
=-=-=-
T_feeds_switch
// RecvrParamFlags:
//   0: ParamFeedsIfOrSwitch
===
{
	"Flags": 0,
	"RecvrParamFlags": [
		8
	],
	"ReturnFlags": []
}
=-=-=-
T_feeds_if_toocomplex
===
{
	"Flags": 0,
	"RecvrParamFlags": [
		0,
		0
	],
	"ReturnFlags": []
}
=-=-=-
T_feeds_if_redefined
===
{
	"Flags": 0,
	"RecvrParamFlags": [
		0
	],
	"ReturnFlags": []
}
=-=-=-
T_feeds_if_redefined2
===
{
	"Flags": 0,
	"RecvrParamFlags": [
		0
	],
	"ReturnFlags": []
}
=-=-=-
T_feeds_multi_if
// RecvrParamFlags:
//   0: ParamFeedsIfOrSwitch
//   1: ParamNoInfo
===
{
	"Flags": 0,
	"RecvrParamFlags": [
		8,
		0
	],
	"ReturnFlags": []
}
=-=-=-
T_feeds_if_redefined_indirectwrite
===
{
	"Flags": 0,
	"RecvrParamFlags": [
		0
	],
	"ReturnFlags": []
}
=-=-=-
T_feeds_if_redefined_indirectwrite_copy
===
{
	"Flags": 0,
	"RecvrParamFlags": [
		0
	],
	"ReturnFlags": []
}
=-=-=-
T_feeds_if_expr1
// RecvrParamFlags:
//   0: ParamFeedsIfOrSwitch
===
{
	"Flags": 0,
	"RecvrParamFlags": [
		8
	],
	"ReturnFlags": []
}
=-=-=-
T_feeds_if_expr2
===
{
	"Flags": 0,
	"RecvrParamFlags": [
		0
	],
	"ReturnFlags": []
}
=-=-=-
T_feeds_if_expr3
===
{
	"Flags": 0,
	"RecvrParamFlags": [
		0
	],
	"ReturnFlags": []
}
=-=-=-
T_feeds_if_shift_may_panic
===
{
	"Flags": 0,
	"RecvrParamFlags": [
		0
	],
	"ReturnFlags": [
		0
	]
}
=-=-=-
T_feeds_if_maybe_divide_by_zero
===
{
	"Flags": 0,
	"RecvrParamFlags": [
		0
	],
	"ReturnFlags": []
}
=-=-=-
T_feeds_indcall
// RecvrParamFlags:
//   0: ParamFeedsIndirectCall
===
{
	"Flags": 0,
	"RecvrParamFlags": [
		4
	],
	"ReturnFlags": []
}
=-=-=-
T_feeds_indcall_and_if
// RecvrParamFlags:
//   0: ParamFeedsIndirectCall|ParamFeedsIfOrSwitch
===
{
	"Flags": 0,
	"RecvrParamFlags": [
		12
	],
	"ReturnFlags": []
}
=-=-=-
T_feeds_indcall_with_copy
// RecvrParamFlags:
//   0: ParamFeedsIndirectCall
===
{
	"Flags": 0,
	"RecvrParamFlags": [
		4
	],
	"ReturnFlags": []
}
=-=-=-
T_feeds_interface_method_call
// RecvrParamFlags:
//   0: ParamFeedsInterfaceMethodCall
===
{
	"Flags": 0,
	"RecvrParamFlags": [
		2
	],
	"ReturnFlags": []
}
=-=-=-
init
===
{
	"Flags": 0,
	"RecvrParamFlags": [],
	"ReturnFlags": []
}
=-=-=-
